sup3r/preprocessing/batch_handling.py

```diff
@@ -991,47 +1028,57 @@ def __next__(self):
+        high_res = None
 
-                if high_res is None:
-                    hr_shape = (self.batch_size,) + obs_daily_avg.shape
-                    high_res = np.zeros(hr_shape, dtype=np.float32)
+        for i in range(self.batch_size):
+            _, obs_daily_avg = handler.get_next()
+            self.current_batch_indices.append(handler.current_obs_index)
 
-                    msg = ('SpatialBatchHandlerCC can only use n_temporal==1 '
-                           'but received HR shape {} with n_temporal={}.'
-                           .format(hr_shape, hr_shape[3]))
-                    assert hr_shape[3] == 1, msg
+            if high_res is None:
+                hr_shape = (self.batch_size,) + obs_daily_avg.shape
+                high_res = np.zeros(hr_shape, dtype=np.float32)
 
-                high_res[i] = obs_daily_avg
+                msg = ('SpatialBatchHandlerCC can only use n_temporal==1 '
+                       'but received HR shape {} with n_temporal={}.'
+                       .format(hr_shape, hr_shape[3]))
+                assert hr_shape[3] == 1, msg
 
-            low_res = spatial_coarsening(high_res, self.s_enhance)
-            low_res = low_res[:, :, :, 0, :]
-            high_res = high_res[:, :, :, 0, :]
+            high_res[i] = obs_daily_avg
 
-            high_res = self.BATCH_CLASS.reduce_features(
-                high_res, self.output_features_ind)
+        low_res = spatial_coarsening(high_res, self.s_enhance)
+        low_res = low_res[:, :, :, 0, :]
+        high_res = high_res[:, :, :, 0, :]
 
-            if (self.output_features is not None
-                    and 'clearsky_ratio' in self.output_features):
-                i_cs = self.output_features.index('clearsky_ratio')
-                if np.isnan(high_res[..., i_cs]).any():
-                    high_res[..., i_cs] = nn_fill_array(high_res[..., i_cs])
+        high_res = self.BATCH_CLASS.reduce_features(
+            high_res, self.output_features_ind)
 
-            batch = self.BATCH_CLASS(low_res, high_res)
+        if (self.output_features is not None
+                and 'clearsky_ratio' in self.output_features):
+            i_cs = self.output_features.index('clearsky_ratio')
+            if np.isnan(high_res[..., i_cs]).any():
+                high_res[..., i_cs] = nn_fill_array(high_res[..., i_cs])
 
-            self._i += 1
-            return batch
-        else:
-            raise StopIteration
+        if self.smoothing is not None:
+            feat_iter = [j for j in range(low_res.shape[-1])
+                         if self.training_features[j]
+                         not in self.smoothing_ignore]
+            for i in range(low_res.shape[0]):
+                for j in feat_iter:
+                    low_res[i, ..., j] = gaussian_filter(low_res[i, ..., j],
+                                                         self.smoothing,
+                                                         mode='nearest')
+
```

Comment:
Yeah I should have noted the hard coded method before. I assume nearest sorted some of the boundary issues?

Reply:
Yeah so constant just pads the boundaries with 0 (default) so you get a low-valued border. Nearest does a NN extrapolation so the boundary is seamless.